test/external-modules/jvm-crash/src/javaRestTest/java/org/elasticsearch/test/jvm_crash/JvmCrashIT.java

```diff
@@ -0,0 +1,178 @@
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.jvm_crash;
+
+import org.apache.lucene.util.Constants;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.core.PathUtils;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.AbstractLocalClusterSpecBuilder;
+import org.elasticsearch.test.cluster.local.DefaultEnvironmentProvider;
+import org.elasticsearch.test.cluster.local.DefaultLocalClusterFactory;
+import org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.DefaultSettingsProvider;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.cluster.local.distribution.LocalDistributionResolver;
+import org.elasticsearch.test.cluster.local.distribution.ReleasedDistributionResolver;
+import org.elasticsearch.test.cluster.local.distribution.SnapshotDistributionResolver;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.hamcrest.Matcher;
+import org.junit.AfterClass;
+import org.junit.ClassRule;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.matchesRegex;
+import static org.hamcrest.Matchers.not;
+
+public class JvmCrashIT extends ESRestTestCase {
+
+    private static class StdOutCatchingClusterBuilder extends AbstractLocalClusterSpecBuilder<ElasticsearchCluster> {
+
+        private StdOutCatchingClusterBuilder() {
+            this.settings(new DefaultSettingsProvider());
+            this.environment(new DefaultEnvironmentProvider());
+        }
+
+        @Override
+        public ElasticsearchCluster build() {
+            // redirect stdout before the nodes start up
+            // they are referenced directly by ProcessUtils, so can't be changed afterwards
+            redirectStdout();
+
+            return new DefaultLocalElasticsearchCluster<>(
+                this::buildClusterSpec,
+                new DefaultLocalClusterFactory(
+                    new LocalDistributionResolver(new SnapshotDistributionResolver(new ReleasedDistributionResolver()))
+                )
+            );
+        }
+    }
+
+    private static PrintStream originalOut;
+    private static ByteArrayOutputStream stdOutput;
+
+    private static void redirectStdout() {
+        if (originalOut == null) {
+            originalOut = System.out;
+            stdOutput = new ByteArrayOutputStream();
```

Comment:
Why can't we read the output from disk? The test nodes have an es.out file that should contain the crash info?

Reply:
The JVM crash is only output to standard out (JDK knows nothing about log4j), the `test-cluster.log` file doesn't have it

Reply:
es.out is not log4j, it's the stdout of the ES process.

Reply:
oof, I am mixing up testclusters and the new test framework. For things like `gradlew run` we have `es.out`, but it looks like the new cluster test framework starts the ES processes with inherited IO. 

@mark-vieira is there a strong reason why the stdout/stderr is not captured via a file?

Reply:
We redirect stdout from the ES proccess to stdout in the test process so the crash should be captured in the test logs but perhaps not in a way that can be asserted on. If we need a general way to redirect logs to some other consumer we can certainly add an API for this use case. I think we've used the `getNodeLog()` API for these use cases in the past but that obviously doesn't work for JVM level stuff that doesn't go through log4j.

Reply:
And also it's very rare to have something that goes to stdout/stderr that does not go through log4j